### DISCUSSION TITLE
How to add information to the info.plist on macOS when flet pack

### DISCUSSION THREAD

**Discussion by Francois-Belanger** [flet-dev/flet#3827] [Q&A]
Hi, can you please provide info on how to be able to add more info to the info.plist please ?  I'm using the Bluetooth and need to have the permission setup in the file:

I tried to change the bundle... Signing is not good anymore.
I tried changing the file macOS_utils.py to include this:
    pl["Privacy - Bluetooth Peripheral Usage Description"] = "Can this app communicate with Bluetooth peripherals ?"
    pl["Privacy - Bluetooth Always Usage Description"] = "Can this app use Bluetooth ?"

Still not visible inside the info.plist.

Help please :-)

Thanks
Francois

**Comment by FeodorFitsner** [comment]
So, I understand you need to add that permission to the "inner" app - Flet viewer, not an "outer" shell app generated by PyInstaller?

I don't know why modifying `macos_utils.py` didn't work - seems the right place to go. Do you see "Updating Flet View plist" in the log? Also, not sure if the keys you are using are right - does it allow spaces in keys?

**Comment by Francois-Belanger** [comment]
Maybe I'm missing something :-)

in the dist folder I have an main.app (Application) and main (Unix Executable File)

If I modify the info.plist in the app bundle it doesn't crash anymore on the Bluetooth.  But the bluetooth doesn't work ... maybe it's normal ? 

But if I run the unix executable, I get a message the first time asking if the Terminal app can access the Bluetooth, then this works fine :-)

So... Why :-) 

Thanks and by the way I love Flet...
Francois

**Comment by FeodorFitsner** [comment]
When you first run an app bundle on macOS it unpacks "Flet View" app, which is also a bundle, to `~/.flet` directory. Would it work if you update plist there?

**Comment by Francois-Belanger** [comment]
Just did, still don't have the bluetooth working with the bundle.  If I modify the info.plist of the bundle... the app signing is now outdated right ?  Could this affect permissions ?

Also, if I run the Unix executable, it open the Terminal and I can see my logs... But when I use the bundle, I can't find where the logs (print) are ?

Thanks
Francois

**Comment by Francois-Belanger** [comment]
@FeodorFitsner can you please give me your opinion on this please :-)

Thanks

**Comment by FeodorFitsner** [comment]
I don't know... modifying `info.plist` of the app bundle inside `~/.flet` is the correct way to go.

You can re-sign updated bundle with the following command:

```
codesign -s - --force --all-architectures --timestamp --deep <path-to-bundle>
```
